core/server/api_container/server/service_network/default_service_network.go

```diff
@@ -841,87 +842,95 @@ func (network *DefaultServiceNetwork) UnpauseService(ctx context.Context, servic
 	return nil
 }
 
-func (network *DefaultServiceNetwork) ExecCommand(ctx context.Context, serviceIdentifier string, command []string) (int32, string, error) {
+func (network *DefaultServiceNetwork) RunExec(ctx context.Context, serviceIdentifier string, userServiceCommand []string) (*exec_result.ExecResult, error) {
 	// NOTE: This will block all other operations while this command is running!!!! We might need to change this so it's
 	// asynchronous
 	network.mutex.Lock()
 	defer network.mutex.Unlock()
 
 	serviceRegistration, err := network.getServiceRegistrationForIdentifierUnlocked(serviceIdentifier)
 	if err != nil {
-		return 0, "", stacktrace.Propagate(err, "An error occurred while getting service registration for identifier '%v'", serviceIdentifier)
+		return nil, stacktrace.Propagate(err, "An error occurred while getting service registration for identifier '%v'", serviceIdentifier)
 	}
 
-	// NOTE: This is a SYNCHRONOUS command, meaning that the entire network will be blocked until the command finishes
-	// In the future, this will likely be insufficient
-
 	serviceUuid := serviceRegistration.GetUUID()
-	userServiceCommand := map[service.ServiceUUID][]string{
-		serviceUuid: command,
+	userServiceCommands := map[service.ServiceUUID][]string{
+		serviceUuid: userServiceCommand,
 	}
 
 	successfulExecCommands, failedExecCommands, err := network.kurtosisBackend.RunUserServiceExecCommands(
-		ctx,
-		network.enclaveUuid,
-		userServiceCommand)
+		ctx, network.enclaveUuid, userServiceCommands)
 	if err != nil {
-		return 0, "", stacktrace.Propagate(
+		return nil, stacktrace.Propagate(
 			err,
-			"An error occurred calling kurtosis backend to exec command '%v' against service '%v'",
-			command,
+			"An error occurred running exec command '%v' against service '%v'",
+			userServiceCommand,
 			serviceIdentifier)
 	}
-	if len(failedExecCommands) > 0 {
-		serviceExecErrs := []string{}
-		for serviceUUID, err := range failedExecCommands {
-			wrappedErr := stacktrace.Propagate(
-				err,
-				"An error occurred attempting to run a command in a service with UUID `%v'",
-				serviceUUID,
-			)
-			serviceExecErrs = append(serviceExecErrs, wrappedErr.Error())
-		}
-		return 0, "", stacktrace.NewError(
-			"One or more errors occurred attempting to exec command(s) in the service(s): \n%v",
-			strings.Join(
-				serviceExecErrs,
-				"\n\n",
-			),
-		)
+
+	if execResult, found := successfulExecCommands[serviceUuid]; found {
```

Comment:
In this case this would be a Kurtosis internal bug, and I can't really find a reason why we would end up in this situation. But in the code I replaced before, we were doing something similar, so I guess that guided me toward being extra conservative. I would be fine throwing an error here. Again, this should never happen

To your second question, this would be hit with the following maps returned by the backend:
```
successfulExecCommands -> {
    serviceUuid -> ExecResult(...)
},
failedExecCommands -> {
    [serviceUuid OR someOtherRandomUuidOutOfNowhere] -> err
}
```

whereas the second clause line 880 would be hit with:
```
successfulExecCommands -> {},
failedExecCommands -> {}
```

So, slightly different, but overall both would indicate a pretty serious and unexpected bug, so I'll fail in this case as well

Reply:
Sounds good to me :)